libr/core/linux_heap_glibc.c

```diff
@@ -303,19 +337,43 @@ static void GH(print_arena_stats)(RCore *core, GHT m_arena, MallocState *main_ar
 }
 
 static bool GH(r_resolve_main_arena)(RCore *core, GHT *m_arena) {
+
 	if (!core || !core->dbg || !core->dbg->maps) {
```

Comment:
use r_return_val_if_fail instead

Reply:
not having dbg.maps is fine, so just chk for core and coredbg with the returniffail and keep the dbg->maps with a simple if

Reply:
I use dbg->maps to find the libc, so It's required to check that is not empty, or not ?